File: backend/app/schemas/node.py
"""节点与相册相关 DTO。"""

from pydantic import BaseModel, Field


class NodeResponse(BaseModel):
    id: int
    parent_id: int | None
    name: str
    path: str
    node_type: str
    source_type: str
    image_count: int
    subdir_count: int
    archive_count: int = 0
    cover_rel_path: str | None = None
    cover_manual: bool = False
    dir_mtime: float | None = None
    created_at: float | None = None

    model_config = {"from_attributes": True}


class RecentNodesResponse(BaseModel):
    total: int
    items: list[NodeResponse]


class ImageItem(BaseModel):
    index: int
    filename: str


class ImageListResponse(BaseModel):
    node_id: int
    total: int
    items: list[ImageItem]


class CoverCandidate(BaseModel):
    value: str
    label: str
    source_node_id: int


class CoverCandidateListResponse(BaseModel):
    node_id: int
    items: list[CoverCandidate]


class NodeUpdate(BaseModel):
    node_type: str | None = None
    cover_rel_path: str | None = None
    cover_index: int | None = Field(default=None, ge=0)
    cover_manual: bool | None = None


class NodeBatchDelete(BaseModel):
    ids: list[int] = Field(min_length=1)


class NodeBatchDeleteResponse(BaseModel):
    deleted: int
    errors: list[str] = []


class NodeMove(BaseModel):
    ids: list[int] = Field(min_length=1)
    target_parent_id: int | None = None


class NodeMoveResponse(BaseModel):
    moved: int
    errors: list[str] = []

File: frontend/src/types/node.ts
export type NodeItem = {
  id: number
  parent_id: number | null
  name: string
  path: string
  node_type: string
  source_type: string
  image_count: number
  subdir_count: number
  archive_count: number
  cover_rel_path: string | null
  cover_manual: boolean
  dir_mtime?: number | null
  created_at?: number | null
}

export type CoverCandidate = {
  value: string
  label: string
  source_node_id: number
}

export type CoverCandidateList = {
  node_id: number
  items: CoverCandidate[]
}

export type ImageItem = {
  index: number
  filename: string
}

export type NodeUpdate = {
  node_type?: string
  cover_rel_path?: string | null
  cover_index?: number
  cover_manual?: boolean
}

export type ImageList = {
  node_id: number
  total: number
  items: ImageItem[]
}

export type NodeBatchDeleteResult = {
  deleted: number
  errors: string[]
}

export type NodeMovePayload = {
  ids: number[]
  target_parent_id: number | null
}

export type NodeMkdirPayload = {
  parent_id?: number | null
  name: string
}

export type NodeMkdirResult = {
  id: number
  path: string
  name: string
}

export type NodeMoveResult = {
  moved: number
  errors: string[]
}

export type ScanJob = {
  id: number
  status: string
  started_at: number | null
  finished_at: number | null
  added: number
  updated: number
  removed: number
  message: string | null
}

export type RecentNodesResult = {
  total: number
  items: NodeItem[]
}
